src/box/lua/config/applier/credentials.lua

```diff
@@ -493,6 +493,28 @@ local function apply(config)
         return
     end
 
+    -- Now the credentials config should be enriched with empty configs
+    -- for roles and users that are present by default on every instance:
+    -- - - super
+    --   - public
+    --   - replication
+    -- - - guest
+    --   - admin
+    --
+    -- It is done to revoke all non-default privileges for this roles and
+    -- users if they are not present in config. Otherwise, the privileges
+    -- will be synced as usual.
```

Comment:
Nits:

1. 'Should be enrihed with empty config' -- I would add somewhere a note that it is later updated with defaults in `privileges_add_defaults()`.
2. 'Otherwise, the privileges will be synced as usual' -- does it mean 'as for a user/role that doesn't exist by default'? I didn't get it initially. I propose to say 'Otherwise, the privileges will be left without changes. It is exactly how it works for all user defined users and roles', or something similar.

Reply:
Rephrased the whole comment:
```lua
      -- Tarantool has the following roles and users present by default on every
      -- instance:
      --
      -- Default roles:
      -- * super
      -- * public
      -- * replication
      --
      -- Default users:
      -- * guest
      -- * admin                                                                                                                       
      --                                                                                                                               
      -- These roles and users have according privileges pre-granted by design.                                                        
      -- Credentials applier adds such privileges with `priviliges_add_default()`                                                      
      -- when syncing. So, for the excessive (non-default) privs to be removed,                                                        
      -- these roles and users must be present inside configuration at least in a                                                      
      -- form of an empty table. Otherwise, the privileges will be left unchanged,                                                     
      -- similar to all used-defined roles and users.                                                                                  
```